src/modules/personenkontext/domain/personenkontext-workflow.ts

```diff
@@ -0,0 +1,344 @@
+import { OrganisationsTyp } from '../../organisation/domain/organisation.enums.js';
+import { OrganisationMatchesRollenart } from '../specification/organisation-matches-rollenart.js';
+import { PersonPermissions } from '../../authentication/domain/person-permissions.js';
+import { OrganisationID } from '../../../shared/types/aggregate-ids.types.js';
+import { RollenSystemRecht } from '../../rolle/domain/rolle.enums.js';
+import { DomainError } from '../../../shared/error/domain.error.js';
+import { EntityNotFoundError } from '../../../shared/error/entity-not-found.error.js';
+import { RolleNurAnPassendeOrganisationError } from '../specification/error/rolle-nur-an-passende-organisation.js';
+import { MissingPermissionsError } from '../../../shared/error/missing-permissions.error.js';
+import { PersonenkontexteUpdateError } from './error/personenkontexte-update.error.js';
+import { Personenkontext } from './personenkontext.js';
+import { PersonenkontexteUpdate } from './personenkontexte-update.js';
+import { DbiamPersonenkontextFactory } from './dbiam-personenkontext.factory.js';
+import { DbiamPersonenkontextBodyParams } from '../api/param/dbiam-personenkontext.body.params.js';
+import { OrganisationRepository } from '../../organisation/persistence/organisation.repository.js';
+import { OrganisationScope } from '../../organisation/persistence/organisation.scope.js';
+import { ScopeOperator } from '../../../shared/persistence/scope.enums.js';
+
+export class PersonenkontextWorkflowAggregate {
+    public selectedOrganisationId?: string;
+
+    public selectedRolleId?: string;
+
+    private constructor(
+        private readonly rolleRepo: RolleRepo,
+        private readonly organisationRepo: OrganisationRepo,
+        private readonly organisationRepository: OrganisationRepository,
+        private readonly dbiamPersonenkontextFactory: DbiamPersonenkontextFactory,
+    ) {}
+
+    public static createNew(
+        rolleRepo: RolleRepo,
+        organisationRepo: OrganisationRepo,
+        organisationRepository: OrganisationRepository,
+        dbiamPersonenkontextFactory: DbiamPersonenkontextFactory,
+    ): PersonenkontextWorkflowAggregate {
+        return new PersonenkontextWorkflowAggregate(
+            rolleRepo,
+            organisationRepo,
+            organisationRepository,
+            dbiamPersonenkontextFactory,
+        );
+    }
+
+    // Initialize the aggregate with the selected Organisation and Rolle
+    public initialize(organisationId?: string, rolleId?: string): void {
+        this.selectedOrganisationId = organisationId;
+        this.selectedRolleId = rolleId;
+    }
+
+    // Finds all SSKs that the admin can see
+    public async findAllSchulstrukturknoten(
+        permissions: PersonPermissions,
+        organisationName: string | undefined,
+        limit?: number,
+    ): Promise<OrganisationDo<true>[]> {
+        const scope: OrganisationScope = new OrganisationScope();
```

Comment:
Scopes haben eine direkte Abhängigkeit auf MikroORM und sollen daher nur in der Persistence-Schicht benutzt werden.